### PR TITLE
MOHAWK: RIVEN: Add keymapper support (continued)

### PR DESCRIPTION
**Pull request by bluegr**
This is based on PR #1990, and includes fixes for the comments made in that pull request.

It's been two weeks without any activity in the other PR, and the changes requested were pretty straightforward, so I decided to continue the work that @ccawley2011 started.

### CODE REVIEW

**Comment by ccawley2011** [comment]
Thanks for sorting this out.

One thing I do want to point out, though, is that Skip and Pause are already handled in the [global keymap](https://github.com/scummvm/scummvm/blob/master/base/main.cpp#L349), which is why I didn't add them in the original PR.

File: engines/mohawk/riven.cpp [hunk 1]
```diff
@@ -151,6 +152,8 @@ Common::Error MohawkEngine_Riven::run() {
 		_cursor = new MacCursorManager("Riven");
 
 	initVars();
+	initKeymap();
```
Comment (bgK) [inline engines/mohawk/riven.cpp]: It seems to me `initKeymap` is meant to be called by the base code when starting the engine, not by the engine itself.

**Comment by bgK** [comment]
Thanks! I've noticed a small issue. Otherwise, looks good to me.

**Comment by bgK** [comment]
> One thing I do want to point out, though, is that Skip and Pause are already handled in the [global keymap](https://github.com/scummvm/scummvm/blob/master/base/main.cpp#L349).

The global keymap having shortcuts that apply only for some engines seems incorrect to me. Users see the key bindings in the remap dialog, notice they don't work and then complain about it :)

File: engines/mohawk/riven.cpp (hunk 1 above)
Comment (bluegr) [reply engines/mohawk/riven.cpp]: I'm confused. When you refer to base code, are you talking about MohawkEngine?
If this is the case, then IMHO it's a bad idea to have a common keymap for all the mohawk subengines, as each one has different hotkeys

Comment (bgK) [reply engines/mohawk/riven.cpp]: Sorry for being unclear, I'm referring to this: https://github.com/scummvm/scummvm/blob/5bda62b12a7216215a3b52c5509f39fd25b13e2f/base/main.cpp#L281

Comment (bluegr) [reply engines/mohawk/riven.cpp]: You are quite right. Fixed.

**Comment by bluegr** [comment]
Since all the issues have been addressed now, this can be merged. Thanks for the feedback!

**Pull request merged**